unblob/handlers/filesystem/squashfs.py

```diff
@@ -216,6 +218,40 @@ class SquashFSv4LEHandler(_SquashFSBase):
     HEADER_STRUCT = "squashfs4_super_block_t"
 
 
+class SquashFSv4BEExtractor(Extractor):
+    def is_avm(self, header: Instance) -> bool:
+        # see https://raw.githubusercontent.com/Freetz/freetz/master/tools/make/squashfs4-host-be/AVM-BE-format.txt
+        return header.bytes_used == header.mkfs_time
+
+    def extract(self, inpath: Path, outdir: Path):
+        struct_parser = StructParser(SquashFSv4LEHandler.C_DEFINITIONS)
+
+        with File.from_path(inpath) as f:
+            header = struct_parser.parse(
+                SquashFSv4LEHandler.HEADER_STRUCT, f, Endian.BIG
+            )
+            if self.is_avm(header):
+                extractor = Command(
+                    "sasquatch-v4be",
+                    "-no-exit-code",
+                    "-f",
+                    "-d",
+                    "{outdir}",
+                    "{inpath}",
+                )
+            else:
+                extractor = Command(
+                    "sasquatch-v4be",
+                    "-be",
+                    "-no-exit-code",
+                    "-f",
+                    "-d",
+                    "{outdir}",
+                    "{inpath}",
+                )
```

Comment:
I would suggest a small refactor here. In this way we can be sure that they are always called with the same arguments:

Note, this is just optional, your solution is also correct.

```suggestion
        commands_args = []

        if not self.is_avm(header):
            commands_args.append("-be")

        commands_args.extend(
            [
                "-no-exit-code",
                "-f",
                "-d",
                "{outdir}",
                "{inpath}",
            ]
        )

        extractor =  Command(self.EXECUTIBLE, *commands_args)
```

Reply:
I prefer your design :) Fixed.